packages/scripts/run-in-package.sh

```
@@ -0,0 +1,29 @@
+WORKSPACE_ROOT="$2"
+
+# Extract package directory from file path
+PACKAGE_DIR=$(echo "$FILE_PATH" | sed -n 's|.*\(packages/[^/]*\).*|\1|p')
+
+if [ -z "$PACKAGE_DIR" ]; then
+    echo "File is not in a packages/ directory"
+    exit 1
+fi
+
+# Change to package directory
+cd "$WORKSPACE_ROOT/$PACKAGE_DIR"
```

Shellcheck (0.10.0)

[warning] Use 'cd ... || exit' or 'cd ... || return' in case cd fails.

(SC2164)